internal/source/git.go

```diff
@@ -98,6 +112,23 @@ func (r *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Resul
 	return &Result{Bundle: bundleFS, ResolvedSource: resolvedSource, State: StateUnpacked}, nil
 }
 
+func (r *Git) getCredential(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (string, string, error) {
+	secret := &corev1.Secret{}
+	err := r.Get(ctx, client.ObjectKey{Namespace: r.SecretNamespace, Name: bundle.Spec.Source.Git.Auth.Secret.Name}, secret)
+	if err != nil {
+		return "", "", err
+	}
+	userName := string(secret.Data["username"])
+	password := string(secret.Data["password"])
+	if password == "" {
+		return "", "", errors.New("data.accessToken is missing")
+	}
+	if userName == "" {
+		userName = "user"
+	}
```

Comment:
What's the idea here? Is it that we can use a dummy value when password is an access token? I wonder if we should just enforce that this secret is specifically a `basic-auth` secret, which means we get the guarantee from the kubernetes API server that `username` and `password` are always set.